packages/vscode-extension/src/extension.ts

```diff
@@ -820,6 +821,7 @@ async function runBackgroundAsyncTasks(
   if (isTeamsFxProject) {
     await handlers.autoOpenProjectHandler();
     await TreeViewManagerInstance.updateTreeViewsByContent();
+    await AzureAccountManager.addStatusChangeEvent();
```

Comment:
addStatusChangeEvent is more about AzureAccountManager itsself. Why do we move it outside?

Reply:
This means statusMap was set before statusChange notify connected. If any azure events happen during this period, it will be missed. What about we doesn't export default in AzureAccountManager and use getInstance whenever needed?

Reply:
Updated to keep event registration and split reading codes into another method.